### PR TITLE
TC_14.002.01 | Header > Search Box | Auto-Completion Suggestion Selection

### PR DESCRIPTION
**Pull request by Infared1**
[https://github.com/RedRoverSchool/JenkinsQA_JS_2024_fall/issues/142](#142)

*sorry, I was forced to delete the previous branch and do everything again because of errors in commits

### CODE REVIEW

**Review by Negrescu29 — APPROVED** [review]
lgtm

File: cypress/e2e/headerSearchBox.cy.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/// <reference types="cypress" />
+
+describe('US_14.002 | Header > Search Box', () => {
+
+    beforeEach(() => {
+        cy.get('a.model-link').contains('Dashboard')
```
Comment (adiuzhikov) [inline cypress/e2e/headerSearchBox.cy.js:6]: I am uncertain why you are using the method .contains() within the hook. This method does not perform any action and just locates the element, and there is no further action taken.

File: cypress/e2e/headerSearchBox.cy.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/// <reference types="cypress" />
+
+describe('US_14.002 | Header > Search Box', () => {
+
+    beforeEach(() => {
+        cy.get('a.model-link').contains('Dashboard')
+    })
+    it('TC_14.002.01 | Auto-Completion Suggestion Selection', () => {
+        cy.get('input[id="search-box"]').click().clear()
```
Comment (adiuzhikov) [inline cypress/e2e/headerSearchBox.cy.js:9]: Here you don't have to .clear() the input field, it's empty by default.

File: cypress/e2e/headerSearchBox.cy.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+describe('US_14.002 | Header > Search Box', () => {
+
+    beforeEach(() => {
+        cy.get('a.model-link').contains('Dashboard')
+    })
+    it('TC_14.002.01 | Auto-Completion Suggestion Selection', () => {
+        cy.get('input[id="search-box"]').click().clear()
+        .type('man')
+        cy.get('li').contains('manage').click().should('be.visible')
```
Comment (adiuzhikov) [inline cypress/e2e/headerSearchBox.cy.js:11]: It appears that the logic is incorrect. To determine if the element “li” is visible, you should either remove the .click() method here or after clicking, find another element that contains the “manage” text.

File: cypress/e2e/headerSearchBox.cy.js (hunk 1 above)
Comment (Infared1) [reply cypress/e2e/headerSearchBox.cy.js:6]: Yes, we just need to make sure we are on the right page

File: cypress/e2e/headerSearchBox.cy.js (hunk 2 above)
Comment (Infared1) [reply cypress/e2e/headerSearchBox.cy.js:9]: The lecture said that it is good practice to use .clear() first to make sure the input is empty

**Review by Tarantata — APPROVED** [review]

**Pull request merged**